pkg/webhook/server/tls/file_source_test.go

```diff
@@ -56,8 +58,10 @@ func TestFileSource_ReadsFile(t *testing.T) {
 		Log:            logtesting.TestLogger{T: t},
 	}
 	stopCh := make(chan struct{})
-	defer close(stopCh)
```

Comment:
Question: If we don't defer closing the stop channel, will it not be the case that the `source.Run` continues running in a goroutine if we fail the test i.e on line 69? Same for the change on line 100.

Reply:
Question2: Would it perhaps make sense to replace `time.Sleep` on line 78 with the `errGroup.Wait..` somehow? I understand that by the time we have finished sleeping we already expect whatever `source.Run` does to be finished?

Reply:
I thought t.Fatal didn't run deferred code, but I was wrong. Added more closes (shame about defer / wait being incompatible)

The goroutine this is actually testing re-reads a file from disk every `interval` forever, so the sleep can't be replaced. It could be refactored so you can pass it a fake `time.Ticker` to make it testable but there's not a huge amount of benefit as you'd still have to wait an arbitrary amount of time for the disk read to happen.

Reply:
> shame about defer / wait being incompatible

True, I didn't realize that we need to have closed the channel in order to check for the error. I guess we could have put that goroutine error check itself into the `defer` block together with the `close`,  but at the end that's probably not worth the effort at this point.